src/coreComponents/dataRepository/xmlWrapper.hpp

```diff
@@ -46,21 +46,225 @@ namespace geos
  */
 namespace xmlWrapper
 {
-/// Alias for the type of xml document.
-using xmlDocument = pugi::xml_document;
 
 /// Alias for the type of the result from an xml parse attempt.
 using xmlResult = pugi::xml_parse_result;
 
 /// Alias for the type of an xml node.
+/// An xmlNode behave as a pointer object: passing it by value to a function which modify it
+/// will modify the original xmlNode object.
 using xmlNode = pugi::xml_node;
 
 /// Alias for the type of an xml attribute.
+/// An xmlAttribute behave as a pointer object: passing it by value to a function which modify it
+/// will modify the original xmlAttribute object.
 using xmlAttribute = pugi::xml_attribute;
 
 /// Alias for the type variant of an xml node.
 using xmlTypes = pugi::xml_node_type;
```

Comment:
Depending on where you want to push your wrapping, I would expect all of these types to be wrapped as well?

Reply:
Indeed, I planned to wrap these types in a later PR.